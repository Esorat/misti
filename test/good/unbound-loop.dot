digraph "unbound-loop" {
    node [shape=box];
    subgraph "cluster_test1" {
        label="test1";
        "test1_151" [label="let a: Int = 20"];
        "test1_152" [label="while (a > 10)"];
        "test1_155" [label="return a",style=filled,fillcolor="#66A7DB"];
        "test1_151" -> "test1_152";
        "test1_152" -> "test1_152";
        "test1_152" -> "test1_155";
    }
    subgraph "cluster_test2" {
        label="test2";
        "test2_157" [label="let a: Int = 20"];
        "test2_158" [label="until (a > 10)"];
        "test2_161" [label="return a",style=filled,fillcolor="#66A7DB"];
        "test2_157" -> "test2_158";
        "test2_158" -> "test2_158";
        "test2_158" -> "test2_161";
    }
    subgraph "cluster_test3" {
        label="test3";
        "test3_163" [label="let a: Int = 20"];
        "test3_164" [label="while (a > 10)"];
        "test3_166" [label="a += 1"];
        "test3_169" [label="return a",style=filled,fillcolor="#66A7DB"];
        "test3_163" -> "test3_164";
        "test3_164" -> "test3_166";
        "test3_166" -> "test3_164";
        "test3_164" -> "test3_169";
    }
    subgraph "cluster_test4" {
        label="test4";
        "test4_171" [label="let a: Int = 10"];
        "test4_172" [label="repeat (a)"];
        "test4_175" [label="return a",style=filled,fillcolor="#66A7DB"];
        "test4_171" -> "test4_172";
        "test4_172" -> "test4_172";
        "test4_172" -> "test4_175";
    }
    subgraph "cluster_test5" {
        label="test5";
        "test5_177" [label="let i: Int = 0"];
        "test5_178" [label="while (i > 0)"];
        "test5_177" -> "test5_178";
        "test5_178" -> "test5_178";
    }
    subgraph "cluster_test6" {
        label="test6";
        "test6_181" [label="let i: Int = 0"];
        "test6_182" [label="while (10 < i)"];
        "test6_181" -> "test6_182";
        "test6_182" -> "test6_182";
    }
    subgraph "cluster_test7" {
        label="test7";
        "test7_185" [label="let i: Bool = false"];
        "test7_186" [label="while (i)"];
        "test7_185" -> "test7_186";
        "test7_186" -> "test7_186";
    }
    subgraph "cluster_test8" {
        label="test8";
        "test8_189" [label="let i: Bool = true"];
        "test8_190" [label="while (!i)"];
        "test8_189" -> "test8_190";
        "test8_190" -> "test8_190";
    }
    subgraph "cluster_fn" {
        label="fn";
    }
    subgraph "cluster_test9" {
        label="test9";
        "test9_193" [label="let a: Int = 20"];
        "test9_194" [label="while (a > 10)"];
        "test9_196" [label="fn(a)"];
        "test9_199" [label="return a",style=filled,fillcolor="#66A7DB"];
        "test9_193" -> "test9_194";
        "test9_194" -> "test9_196";
        "test9_196" -> "test9_194";
        "test9_194" -> "test9_199";
    }
    subgraph "cluster_Test__receive_internal_comment_1790_test" {
        label="Test__receive_internal_comment_1790_test";
        "Test__receive_internal_comment_1790_test_201" [label="let s: S = S{a: 1}"];
        "Test__receive_internal_comment_1790_test_202" [label="let i: Int = 0"];
        "Test__receive_internal_comment_1790_test_204" [label="while (i < s.a)"];
        "Test__receive_internal_comment_1790_test_206" [label="i += 1"];
        "Test__receive_internal_comment_1790_test_201" -> "Test__receive_internal_comment_1790_test_202";
        "Test__receive_internal_comment_1790_test_202" -> "Test__receive_internal_comment_1790_test_204";
        "Test__receive_internal_comment_1790_test_204" -> "Test__receive_internal_comment_1790_test_206";
        "Test__receive_internal_comment_1790_test_206" -> "Test__receive_internal_comment_1790_test_204";
    }
}
