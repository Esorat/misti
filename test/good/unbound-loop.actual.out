[HIGH] UnboundLoops: Found unbounded loop
test/good/unbound-loop.tact:3:12:
  2 |     let a: Int = 20; // read-only variable
> 3 |     while (a > 10) { // ERROR: a is not changing within the loop
                 ^
  4 |     }
The condition variable doesn't change inside the loop
Help: Consider changing the variable inside the loop to ensure it terminates
See: https://nowarp.io/tools/misti/docs/detectors/UnboundLoops

[HIGH] UnboundLoops: Found unbounded loop
test/good/unbound-loop.tact:11:14:
  10 |     do  { // ERROR: a is not changing within the loop
> 11 |     } until (a > 10);
                    ^
  12 |     return a;
The condition variable doesn't change inside the loop
Help: Consider changing the variable inside the loop to ensure it terminates
See: https://nowarp.io/tools/misti/docs/detectors/UnboundLoops

[HIGH] UnboundLoops: Found unbounded loop
test/good/unbound-loop.tact:31:12:
  30 |     let i: Int = 0;
> 31 |     while (i > 0) {} // WARNING: unbound loop
                  ^
  32 | }
The condition variable doesn't change inside the loop
Help: Consider changing the variable inside the loop to ensure it terminates
See: https://nowarp.io/tools/misti/docs/detectors/UnboundLoops

[HIGH] UnboundLoops: Found unbounded loop
test/good/unbound-loop.tact:36:17:
  35 |     let i: Int = 0;
> 36 |     while (10 < i) {} // WARNING: unbound loop
                       ^
  37 | }
The condition variable doesn't change inside the loop
Help: Consider changing the variable inside the loop to ensure it terminates
See: https://nowarp.io/tools/misti/docs/detectors/UnboundLoops

[HIGH] UnboundLoops: Found unbounded loop
test/good/unbound-loop.tact:42:12:
  41 |     let i: Bool = false;
> 42 |     while (i) {} // WARNING: unbound loop
                  ^
  43 | }
The condition variable doesn't change inside the loop
Help: Consider changing the variable inside the loop to ensure it terminates
See: https://nowarp.io/tools/misti/docs/detectors/UnboundLoops

[HIGH] UnboundLoops: Found unbounded loop
test/good/unbound-loop.tact:47:13:
  46 |     let i: Bool = true;
> 47 |     while (!i) {} // WARNING: unbound loop
                   ^
  48 | }
The condition variable doesn't change inside the loop
Help: Consider changing the variable inside the loop to ensure it terminates
See: https://nowarp.io/tools/misti/docs/detectors/UnboundLoops

[MEDIUM] ReadOnlyVariables: Read-only variable
test/good/unbound-loop.tact:30:9:
  29 | fun test5() {
> 30 |     let i: Int = 0;
               ^
  31 |     while (i > 0) {} // WARNING: unbound loop
Help: Consider creating a constant instead
See: https://nowarp.io/tools/misti/docs/detectors/ReadOnlyVariables

[MEDIUM] ReadOnlyVariables: Read-only variable
test/good/unbound-loop.tact:35:9:
  34 | fun test6() {
> 35 |     let i: Int = 0;
               ^
  36 |     while (10 < i) {} // WARNING: unbound loop
Help: Consider creating a constant instead
See: https://nowarp.io/tools/misti/docs/detectors/ReadOnlyVariables

[MEDIUM] ReadOnlyVariables: Read-only variable
test/good/unbound-loop.tact:41:9:
  40 | fun test7() {
> 41 |     let i: Bool = false;
               ^
  42 |     while (i) {} // WARNING: unbound loop
Help: Consider creating a constant instead
See: https://nowarp.io/tools/misti/docs/detectors/ReadOnlyVariables

[MEDIUM] ReadOnlyVariables: Read-only variable
test/good/unbound-loop.tact:46:9:
  45 | fun test8() {
> 46 |     let i: Bool = true;
               ^
  47 |     while (!i) {} // WARNING: unbound loop
Help: Consider creating a constant instead
See: https://nowarp.io/tools/misti/docs/detectors/ReadOnlyVariables

[MEDIUM] ReadOnlyVariables: Read-only variable
test/good/unbound-loop.tact:69:13:
  68 |     receive("test") {
> 69 |         let s: S = S {
                   ^
  70 |             a: 1,
Help: Consider creating a constant instead
See: https://nowarp.io/tools/misti/docs/detectors/ReadOnlyVariables